imagespace/server/__init__.py

```diff
@@ -173,18 +173,43 @@ def load(info):
 def solr_documents_from_field(field, values, classifications=None):
     """Given a field, and a list of values, return list of relevant solr documents.
 
-    This performs several requests, each of size CHUNK_SIZE to avoid sending
-    too much data (HTTP 413).
-
     Additionally it can take an iterable of classifications which will be
     searched for through Solr.
 
     :param paths: List of solr paths corresponding to the Solr id attribute
     :param classifications: List of classifications to search by
     :returns: List of solr documents
     """
-    CHUNK_SIZE = 20
-    documents = []
+    def paged_request(params):
+        """
+        Takes a params dictionary and manages paging.
+
+        Uses POST so very large request bodies can be sent to Solr.
+
+        Returns a list of all documents.
+        """
+        documents = []
+
+        # Adjust paging params
+        params['start'] = 0
+        params['rows'] = 1000
+
+        numFound = None
+        numRetrieved = None
+        while numRetrieved is None or numRetrieved < numFound:
+            r = requests.post(imagespaceSetting.get('IMAGE_SPACE_SOLR') + '/select',
+                              data=params,
+                              verify=False).json()
+
+            numFound = r['response']['numFound']
+            numRetrieved = len(r['response']['docs']) if numRetrieved is None \
+                           else numRetrieved + len(r['response']['docs'])
+            documents += r['response']['docs']
+
+            # Setup offset for next request
+            params['start'] = numRetrieved
```

Comment:
Does solr return a deterministic result list every time for a query? I've been bitten in the past when assuming this and it not happening. I usually have to ask for a sorted response in order for this to work in a stable manner.